presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -238,6 +238,12 @@ default void finishCreateTable(ConnectorSession session, ConnectorOutputTableHan
     }
 
     /**
+     * Start a SELECT/UPDATE/INSERT/DELETE query. This notification is triggered after the planning phase completes.
+     */
+    default void beginQuery(ConnectorSession session)
+    {}
```

Comment:
Fixed in 9073a6e60fe7631f09f8a5194cc73bbcee4f46dd for `beginQuery` and in 6661442f10d481a56e5b3c8e57d7735c1802a1b6 for `endQuery`.